libvast/src/system/partition_transformer.cpp

```diff
@@ -291,41 +340,66 @@ partition_transformer_actor::behavior_type partition_transformer(
     },
     [self](atom::internal, atom::resume, atom::done, vast::id offset) {
       VAST_DEBUG("{} got new offset {}", *self, offset);
-      self->state.data.offset = offset;
-      for (auto& slice : self->state.slices) {
-        slice.offset(offset);
-        offset += slice.rows();
-        self->state.add_slice(slice);
-        self->state.stage->out().push(slice);
+      for (auto& [_, data] : self->state.data) {
+        data.offset = offset;
+        for (auto& slice : self->state.slices) {
+          slice.offset(offset);
+          offset += slice.rows();
+          self->state.add_slice(slice);
+          self->state.stage->out().push(slice);
+        }
+      }
+      for (auto& [layout, typed_indexers] : self->state.indexers)
+        for (auto& [qf, idx] : typed_indexers)
+          self->state.data[layout].indexer_chunks.emplace_back(qf.name(),
+                                                               chunkify(idx));
+      for (auto& [layout, partition_data] : self->state.data) {
+        auto& mutable_synopsis = partition_data.synopsis.unshared();
+        mutable_synopsis.shrink();
+        // TODO: It would probably make more sense if the partition
+        // synopsis keeps track of offset/events internally.
+        mutable_synopsis.offset = partition_data.offset;
+        mutable_synopsis.events = partition_data.events;
```

Comment:
From here to Line 413 we do 3 passes over `self->state.data`. Is there a particular reason not to do a single loop that finishes up one partition on each iteration?

Reply:
I've combined the first two, the last one is still required to be separate because all indexers need to be chunkified already for creating the final flatbuffer.